scarlet/lite/initialization.py

```diff
@@ -318,103 +318,159 @@ def init_fista_component(center, bbox, sed, morph, observation, bg_thresh=None):
     return component
 
 
+class Chi2InitParameters:
+    """Parameters used to initialize all sources with chi^2 detections
+
+    There are a large number of parameters that are universal for all of the
+    sources being initialized from the same set of observed images.
+    To simplify the API those parameters are all initialized by this class
+    and passed to `init_main_source` for each source.
+    It also creates temporary objects that only need to be created once for
+    all of the sources in a blend.
+    """
+    def __init__(self, observation, detect=None, min_snr=50, use_mask=False, percentile=25, thresh=0.5):
+        """Initialize the class
+
+        Parameters
+        ----------
+        observation: `scarlet.lite.LiteObservation`
+            The observation containing the blend
+        detect: `numpy.ndarray`
+            The array that contains a 2D image used for detection.
+        centers: `list` of `tuple`
+            The coordinates of all the peak locations to use for initializing sources.
+        min_snr: `float`
+            The minimum SNR required per component.
+            So a 2-component source requires at least `2*min_snr` while sources
+            with SNR < `min_snr` will be initialized with the PSF.
+        use_mask: `bool`
+            Whether to use the monotonic mask or weighted monotonicity for
+            initialization.
+        percentile: `float`
+            The percentage of the overall central flux to attribute to the disk.
+        thresh: `float`
+            The threshold used to trim the morphology
```

Comment:
Threshold meaning minimum flux below which pixels are set to zero?

Reply:
Good point. I updated the docs to explain that all pixels below `thresh` * `background rms` are set to zero.